### PR TITLE
[ALLUXIO-3120] Fix issue where secondary deletes completed log 

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-3120

### CODE REVIEW

File: core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalLogWriter.java [hunk 1]
```diff
@@ -364,6 +320,35 @@ private void createNewLogFile() throws IOException {
     LOG.info("Created current log file: {}", currentLog);
   }
 
+  private void completeLog(UfsJournalFile currentLog, long nextSequenceNumber) throws IOException {
```
Comment (gpang) [inline core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalLogWriter.java]: Could you add javadoc for this, explaining what this method will do and what the parameters should be?

File: core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalLogWriter.java [hunk 2]
```diff
@@ -364,6 +320,35 @@ private void createNewLogFile() throws IOException {
...
+        LOG.warn("Failed to delete empty journal file {}", current);
+      }
+      return;
+    }
+    LOG.info("Completing log {} with next sequence number {}", current, nextSequenceNumber);
+    String completed = UfsJournalFile
+        .encodeLogFileLocation(mJournal, currentLog.getStart(), nextSequenceNumber).toString();
+
+    if (!mUfs.renameFile(current, completed)) {
+      // Renames could happen concurrently.
```
Comment (gpang) [inline core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalLogWriter.java]: could you add when these completes could happen concurrently?

File: core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalLogWriter.java (hunk 1 above)
Comment (aaudiber) [reply core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalLogWriter.java]: done

File: core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalLogWriter.java (hunk 2 above)
Comment (aaudiber) [reply core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalLogWriter.java]: added to the javadoc

**Comment by aaudiber** [comment]
@gpang Thanks for the review, I've addressed the comments now

**Comment by aaudiber** [comment]
@yupeng9 I'll see if I can add a test in a followup PR. Merging for now to catch RC2

**Pull request merged**